content/en/docs/usage/make_build.md

```diff
@@ -0,0 +1,352 @@
+---
+title: Building with Make
+date: 2020-01-11T14:09:21+09:00
+draft: false
+weight: 409
+---
+
+## Building `app-helloworld` with Make
+
+The Unikraft build system is based on [`Make`](https://www.gnu.org/software/make/) and [`KConfig`](https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html).
+The `kraft` command-line client is just a wrapper over `make`.
+For more control, you can use the `Makefile` and `config.uk` files instead of `kraft.yaml` and `kraft` to configure, build and run your unikernel.
+Note that this is more complex than using `kraft`, since you will have to deal with more options.
+
+We want to build [`app-helloworld`](https://github.com/unikraft/app-helloworld) using `Make`, with no dependency on `kraft`.
+In order to do that, we need to follow the same steps we take when using `kraft`:
+
+1. [Setup the working environment](docs/usage/make_build/#setup-the-working-environment)
+1. [Configure the application](docs/usage/make_build/#configure-the-application)
+1. [Build the application](docs/usage/make_build/#build-the-application)
+1. [Run the application](docs/usage/make_build/#run-the-application)
+1. [Clean the build files](docs/usage/make_build/#clean-the-build-files)
+
+### Setup the Working Environment
+
+To manually set up the build environment for the [helloworld app](https://github.com/unikraft/app-helloworld), we first have to create a specific directory structure:
+
+```text
+my-unikernel/
+	|-- apps/
+	|   `-- app-helloworld          <- helloworld application
+	|-- libs/                       <- additional libraries go here
+	`-- unikraft/                   <- Unikraft core
+```
+
+We can do this by executing the following sequence of commands:
+
+```console
+$ mkdir my-unikernel
+$ cd my-unikernel
+$ git clone https://github.com/unikraft/unikraft
+$ mkdir libs
+$ mkdir apps
+$ cd apps
+$ git clone https://github.com/unikraft/app-helloworld
+$ cd app-helloworld
+```
+
+All following configuration and build operations are done from within the `helloworld` application folder.
+
+### Configure the Application
+
+With the `Makefile` present and the directory structure correctly set up, we are able to configure the unikernel as the next step.
+This will prompt a text-based user interface that will allow us to select the target architecture (e.g., `aarch64` or `x86-64`) and platform (e.g., `KVM` or `Xen`).
+
+```console
+$ make menuconfig
+```
```

Comment:
```suggestion
With the `Makefile` present and the directory structure correctly set up, we are able to configure the unikernel as the next step:

---console
$ make menuconfig
---

This will prompt a text-based user interface that will allow us to select the target architecture (e.g., `aarch64` or `x86-64`) and platform (e.g., `KVM` or `Xen`).
```